open_spiel/games/solitaire.h

```diff
@@ -0,0 +1,1025 @@
+#ifndef THIRD_PARTY_OPEN_SPIEL_GAMES_SOLITAIRE_H
+#define THIRD_PARTY_OPEN_SPIEL_GAMES_SOLITAIRE_H
+
+#include <array>
+#include <memory>
+#include <string>
+#include <vector>
+#include <deque>
+#include <variant>
+#include <any>
+#include <unordered_map>
+#include <set>
+#include <optional>
+#include <tuple>
+
+#include "open_spiel/spiel.h"
+
+// An implementation of klondike solitaire: https://en.wikipedia.org/wiki/Klondike_(solitaire)
+// More specifically, it is K+ solitaire, which allows the player to play any card from the deck/waste that would
+// normally become playable after some number of draws in standard klondike solitaire. For a more in-depth
+// description of K+ solitaire, see http://web.engr.oregonstate.edu/~afern/papers/solitaire.pdf. This implementation
+// also gives rewards at intermediate states like most electronic versions of solitaire do, rather than only at
+// terminal states.
+
+// ANSI color codes
+#define RESET "\033[0m"
+#define RED   "\033[31m"
+#define BLACK "\033[37m"
+#define BLUE  "\033[34m"
+
+// Glyphs & Strings
+#define GLYPH_HIDDEN   "\U0001F0A0"
+#define GLYPH_EMPTY    "\U0001F0BF"
+#define GLYPH_SPADES   "\U00002660"
+#define GLYPH_HEARTS   "\U00002665"
+#define GLYPH_CLUBS    "\U00002663"
+#define GLYPH_DIAMONDS "\U00002666"
+#define GLYPH_ARROW    "\U00002190"
+
+namespace open_spiel::solitaire {
+
+    // Default Game Parameters =========================================================================================
+
+    inline constexpr int    kDefaultPlayers      = 1;
+    inline constexpr int    kDefaultDepthLimit   = 150;
+    inline constexpr bool   kDefaultIsColored    = true;
+
+    // Enumerations ====================================================================================================
+
+    enum SuitType     {
```

Comment:
Constant names are very short. Either use `enum class` so they are scoped by the enum name, or use longer names, e.g. `kSuitSpades`.